web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -75,6 +76,25 @@ export class DataStoreService {
       });
   }
 
+  async deleteLayer(projectId: string, layerId: string) {
+    await this._deleteAllFeaturesInLayer(projectId, layerId);
+    return await this.db
```

Comment:
I guess we need to `return await` here instead of `await` because the caller returns a `Promise`?

Reply:
Correct